Neos.ContentRepository.Core/Classes/Feature/NodeModification/Dto/SerializedPropertyValues.php

```diff
@@ -52,10 +49,14 @@ public static function fromArray(array $propertyValues): self
             if (!is_string($propertyName)) {
                 throw new \InvalidArgumentException(sprintf('Invalid property name. Expected string, got: %s', get_debug_type($propertyName)), 1681326239);
             }
-            if ($propertyValue === null) {
-                // this case means we want to un-set a property.
-                $values[$propertyName] = null;
+            if ($propertyValue === null || $propertyValue instanceof UnsetPropertyValue) {
+                $values[$propertyName] = UnsetPropertyValue::get();
             } elseif (is_array($propertyValue)) {
+                if (array_key_exists('value', $propertyValue) && $propertyValue['value'] === null) {
+                    // legacy
+                    $values[$propertyName] = UnsetPropertyValue::get();
+                    continue;
+                }
```

Comment:
without this upcasting, a replay on older event would lead to:

> SerializedPropertyValue::__construct(): Argument #1 ($value) must be of type ArrayObject|array|string|int|float|bool, null given, called in 

alternatively, we could possibly migrate the events table?

Reply:
solved via `SerializedPropertyValue::create`